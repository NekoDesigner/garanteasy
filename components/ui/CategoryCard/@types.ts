import { PropsWithChildren } from 'react';
export interface ICardProps extends PropsWithChildren {
  style?: object;
  testID?: string;
  onPress?: () => void;
}