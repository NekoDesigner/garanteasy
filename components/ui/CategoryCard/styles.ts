import { StyleSheet } from 'react-native';
import { COLORS, SIZES } from '../../../constants';

const styles = StyleSheet.create({
  container: {
    borderRadius: SIZES.radius.l,
    paddingVertical: SIZES.padding.l,
    paddingHorizontal: SIZES.padding.s,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    gap: SIZES.padding.xs,
    height: 138,
    width: 155,
    backgroundColor: COLORS.blueDarker
  },
  text: {
    color: COLORS.light,
    fontFamily: 'Ubuntu',
    fontSize: SIZES.font.m,
    fontWeight: '700',
    textAlign: 'center'
  },
  icon: {
    width: SIZES.icon.l,
    height: SIZES.icon.l,
    color: COLORS.light,
  }
});

export default styles;
