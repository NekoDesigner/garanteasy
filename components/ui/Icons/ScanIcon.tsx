import * as React from "react";
import Svg, { Path } from "react-native-svg";
import { COLORS } from "../../../constants";
import { IconProps } from "./@types";

const DEFAULT_PROPS = {
  size: 24,
  color: COLORS.light
};

function ScanIcon({ size = DEFAULT_PROPS.size, color = DEFAULT_PROPS.color, ...props }: IconProps) {
  return (
    <Svg
      width={size}
      height={size}
      viewBox={`0 0 ${DEFAULT_PROPS.size} ${DEFAULT_PROPS.size}`}
      fill="none"
      {...props}
    >
      <Path
        d="M2.75 14a.75.75 0 01.75.75v2.5a3.25 3.25 0 003.25 3.25h2.5a.75.75 0 010 1.5h-2.5A4.75 4.75 0 012 17.25v-2.5a.75.75 0 01.75-.75zm18.5 0a.75.75 0 01.75.75v2.5A4.75 4.75 0 0117.25 22h-2.5a.75.75 0 010-1.5h2.5a3.25 3.25 0 003.25-3.25v-2.5a.75.75 0 01.75-.75zm-7-7A2.75 2.75 0 0117 9.75v4.5A2.75 2.75 0 0114.25 17h-4.5A2.75 2.75 0 017 14.25v-4.5A2.75 2.75 0 019.75 7h4.5zm-4.5 1.5c-.69 0-1.25.56-1.25 1.25v4.5c0 .69.56 1.25 1.25 1.25h4.5c.69 0 1.25-.56 1.25-1.25v-4.5c0-.69-.56-1.25-1.25-1.25h-4.5zM9.25 2a.75.75 0 010 1.5h-2.5A3.25 3.25 0 003.5 6.75v2.5a.75.75 0 01-1.5 0v-2.5A4.75 4.75 0 016.75 2h2.5zm8 0A4.75 4.75 0 0122 6.75v2.5a.75.75 0 01-1.5 0v-2.5a3.25 3.25 0 00-3.25-3.25h-2.5a.75.75 0 010-1.5h2.5z"
        fill={color}
      />
    </Svg>
  );
}

export default ScanIcon;
