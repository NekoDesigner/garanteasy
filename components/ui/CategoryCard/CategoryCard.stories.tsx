import React from "react";
import { View, Text } from "react-native";
import FashionIcon from "../Icons/FashionIcon";
import GardenIcon from "../Icons/GardenIcon";
import MultimediaIcon from "../Icons/MultimediaIcon";
import ScanIcon from "../Icons/ScanIcon";
import CategoryCard from "./index";

export default {
  title: 'Card',
  component: CategoryCard
};

export const Default = () => (
    <View style={{ flex: 1, flexDirection: 'row', justifyContent: 'space-between', alignItems: 'flex-start', padding: 16, gap: 24, flexWrap: 'wrap' }}>
      <CategoryCard>
        <Text>Jardin</Text>
        <GardenIcon />
    </CategoryCard>
    <CategoryCard>
        <Text>Multimédia</Text>
        <MultimediaIcon />
    </CategoryCard>
    <CategoryCard>
        <Text>Mode</Text>
        <FashionIcon />
    </CategoryCard>
    <CategoryCard>
        <Text>Catégorie</Text>
        <ScanIcon />
    </CategoryCard>
    </View>
);
