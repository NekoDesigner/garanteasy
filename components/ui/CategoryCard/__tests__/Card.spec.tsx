import { render } from "@testing-library/react-native";
import React from "react";
import { Text } from "react-native";
import ScanIcon from "../../Icons/ScanIcon";
import CategoryCard from "../index";
import CardStyles from "../styles";

describe("Card Components", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

    test('renders correctly', () => {
      const tree = render(<CategoryCard>
        <Text>Test Card</Text>
        <ScanIcon />
      </CategoryCard>).toJSON();
      expect(tree).toMatchSnapshot();
    });

  it('renders with custom styles', () => {
    const customStyle = { backgroundColor: 'red' };
    const { getByTestId } = render(
      <CategoryCard style={customStyle} testID="custom-card">
        <Text>Custom Styled Card</Text>
      </CategoryCard>
    );
    const card = getByTestId('custom-card');
    expect(card.props.style).toContainEqual(customStyle);
  });

  it('renders with default testID', () => {
    const { getByTestId } = render(
      <CategoryCard>
        <Text>Default TestID Card</Text>
      </CategoryCard>
    );
    const card = getByTestId('card');
    expect(card).toBeTruthy();
  });

  it('applies text styles to Text children', () => {
    const { getByText } = render(
      <CategoryCard>
        <Text>Styled Text</Text>
      </CategoryCard>
    );
    const textElement = getByText('Styled Text');
    expect(textElement.props.style).toContainEqual(CardStyles.text);
  });
});