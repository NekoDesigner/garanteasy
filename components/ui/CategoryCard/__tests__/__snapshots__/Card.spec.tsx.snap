// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Components renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#050426",
        "borderRadius": 18,
        "display": "flex",
        "flexDirection": "column",
        "gap": 8,
        "height": 138,
        "justifyContent": "center",
        "paddingHorizontal": 14,
        "paddingVertical": 24,
        "width": 155,
      },
      undefined,
    ]
  }
  testID="card"
>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "Ubuntu",
          "fontSize": 16,
          "fontWeight": "700",
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Test Card
  </Text>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={32}
    bbWidth={32}
    fill="none"
    focusable={false}
    height={32}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          {
            "color": "#FFFFFF",
            "height": 32,
            "width": 32,
          },
          undefined,
        ],
        {
          "flex": 0,
          "height": 32,
          "width": 32,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={32}
  >
    <RNSVGGroup
      color="#FFFFFF"
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M2.75 14a.75.75 0 01.75.75v2.5a3.25 3.25 0 003.25 3.25h2.5a.75.75 0 010 1.5h-2.5A4.75 4.75 0 012 17.25v-2.5a.75.75 0 01.75-.75zm18.5 0a.75.75 0 01.75.75v2.5A4.75 4.75 0 0117.25 22h-2.5a.75.75 0 010-1.5h2.5a3.25 3.25 0 003.25-3.25v-2.5a.75.75 0 01.75-.75zm-7-7A2.75 2.75 0 0117 9.75v4.5A2.75 2.75 0 0114.25 17h-4.5A2.75 2.75 0 017 14.25v-4.5A2.75 2.75 0 019.75 7h4.5zm-4.5 1.5c-.69 0-1.25.56-1.25 1.25v4.5c0 .69.56 1.25 1.25 1.25h4.5c.69 0 1.25-.56 1.25-1.25v-4.5c0-.69-.56-1.25-1.25-1.25h-4.5zM9.25 2a.75.75 0 010 1.5h-2.5A3.25 3.25 0 003.5 6.75v2.5a.75.75 0 01-1.5 0v-2.5A4.75 4.75 0 016.75 2h2.5zm8 0A4.75 4.75 0 0122 6.75v2.5a.75.75 0 01-1.5 0v-2.5a3.25 3.25 0 00-3.25-3.25h-2.5a.75.75 0 010-1.5h2.5z"
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
